### ISSUE TITLE
Pylint 4.0.1 error C0103 that started appearing after upgrade to pylint 4

### ISSUE THREAD

**Issue by baruchoxman** [pylint-dev/pylint#10675]
### Bug description

```python
_SIMPLE_VALID_WORKSPACE_YAML_CREATION_RESULT = {
    "name": "new-workspace-for-test",
    "dwh_connector_name": "custom_connector",
    "branch": "prod",
}
```

### Configuration

```ini

```

### Command used

```shell
pylint mytest.py
```

### Pylint output

```python
mytest.py:17:0: C0103: Variable name "_SIMPLE_VALID_WORKSPACE_YAML_CREATION_RESULT" doesn't conform to snake_case naming style ('([^\\W\\dA-Z][^\\WA-Z]*|_[^\\WA-Z]*|__[^\\WA-Z\\d_][^\\WA-Z]+__)$' pattern) (invalid-name)
```

### Expected behavior

This was working before upgrade to pylint 4

### Pylint version

```shell
pylint 4.0.1
astroid 4.0.1
python 3.12.11
```

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue; I can't reproduce, can you share your configuration in particular the invalid-name regex ?

**Comment by jacobtylerwalls** [comment]
Is `_SIMPLE_VALID_WORKSPACE_YAML_CREATION_RESULT` reassigned later in your code? See the release note.

**Comment by baruchoxman** [comment]
> Is `_SIMPLE_VALID_WORKSPACE_YAML_CREATION_RESULT` reassigned later in your code? See the release note.

Yes, you are right - it was re-assigned later-on (it should not have). Thank you for pointing me to this direction.
